frb_example/benches/rust/src/executor.rs

```diff
@@ -0,0 +1,237 @@
+use std::{
+    panic::UnwindSafe,
+    sync::{Arc, Mutex},
+};
+
+use flutter_rust_bridge::{
+    handler::{
+        self, ErrorHandler, Executor, ReportDartErrorHandler, SimpleHandler, ThreadPoolExecutor,
+    },
+    rust2dart::TaskCallback,
+    IntoDart, MessagePort, WrapInfo,
+};
+
+use crate::api::{Metric, Unit};
+const ERROR_MUTEX_LOCK: &str = "Error on mutex lock";
+
+lazy_static::lazy_static! {
+  static ref METRICS: Arc<Mutex<Vec<Metric>>> = Arc::new(Mutex::new(vec![]));
+}
+
+pub type BenchHandler = SimpleHandler<BenchExecutor, BenchErrorHandler>;
+
+pub trait Metrics {
+    fn metrics(&self) -> Vec<Metric>;
+}
+
+impl Metrics for SimpleHandler<BenchExecutor, BenchErrorHandler> {
+    fn metrics(&self) -> Vec<Metric> {
+        let guard = METRICS.lock().expect(ERROR_MUTEX_LOCK);
+        guard.clone()
+    }
+}
+
+/// record a benchmark metric
+///
+/// all benchmark metrics are stored in [`METRICS`]
+fn record(debug_name_string: &str, elapsed: u64, unit: Unit) {
+    let mut guard = METRICS.lock().expect(ERROR_MUTEX_LOCK);
+    guard.push(Metric {
+        name: debug_name_string.to_string(),
+        value: Some(elapsed),
+        extra: None,
+        unit,
+    });
+}
+
+/// simple newtype pattern
+///
+/// this is the bridge's custom [error handler](http://cjycode.com/flutter_rust_bridge/feature/handler.html#example-report-errors-to-your-backend-in-addition-to-telling-dart).
+#[derive(Clone, Copy)]
+pub struct BenchErrorHandler(ReportDartErrorHandler);
+
+impl Default for BenchErrorHandler {
+    fn default() -> Self {
+        Self(ReportDartErrorHandler)
+    }
+}
+
+/// trait bound required by [`SimpleHandler`]
+impl ErrorHandler for BenchErrorHandler {
+    fn handle_error(&self, port: MessagePort, error: handler::Error) {
+        self.0.handle_error(port, error)
+    }
+
+    fn handle_error_sync(
+        &self,
+        error: flutter_rust_bridge::handler::Error,
+    ) -> flutter_rust_bridge::support::WireSyncReturnStruct {
+        self.0.handle_error_sync(error)
+    }
+}
```

Comment:
Looks like we can use the original error handler?

Reply:
indeed!

Reply:
retired in a80da02